### PR TITLE
Add collection:list

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

This PR implements the `collection:list` method with its unit tests.

### How should this be manually tested?

Clone this branch and run unit tests
`./gradlew test`

When it succeed, compile it

`./gradlew jar`

Initiate another java project by adding the compiled SDK as a dependency.

Then, run Kuzzle, create an index `nyc-open-data`, a `yellow-taxi` collection.
Finally, run this code

```java
import io.kuzzle.sdk.*;
import io.kuzzle.sdk.Options.KuzzleOptions;
import io.kuzzle.sdk.Options.Protocol.WebSocketOptions;
import io.kuzzle.sdk.Protocol.WebSocket;

import java.util.concurrent.ConcurrentHashMap;

public class listCollection {
    private static Kuzzle kuzzle;

    public static void main(String[] args) {
        WebSocketOptions opts = new WebSocketOptions();
        opts.setAutoReconnect(true).setConnectionTimeout(42000);

        try {
            WebSocket ws = new WebSocket("localhost", opts);

            kuzzle = new Kuzzle(ws, (KuzzleOptions) null);

            kuzzle.connect();

           ConcurrentHashMap<String, Object> result = kuzzle.getCollectionController().list("nyc-open- 
           data").get();
           System.out.println(result);
            
        }  catch (Exception e) {
            e.printStackTrace();
        }

        kuzzle.disconnect();
    }
};
```

### CODE REVIEW

File: src/main/java/io/kuzzle/sdk/Options/ListOptions.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   *
+   * @param options
+   */
+  public ListOptions(ListOptions options) {
+    this.from = options.getFrom();
+    this.size = options.getSize();
+  }
+
+  public Integer getFrom() {
+    return from;
```
Comment (Aschen) [inline src/main/java/io/kuzzle/sdk/Options/ListOptions.java:24]: What is returned when the uninitialized value is returned? The request contain a number or null?

Comment (Yoann-Abbes) [reply src/main/java/io/kuzzle/sdk/Options/ListOptions.java:24]: If its uninitialized it returns null

File: doc/3/controllers/collection/list/index.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+public CompletableFuture<ConcurrentHashMap<String, Object>> list(
+      final String index) throws NotConnectedException, InternalException
+```
+
+
+| Arguments | Type                   | Description   |
+| --------- | ---------------------- | ------------- |
+| `index`   | <pre>String</pre>      | Index name    |
+| `options` | <pre>ListOptions</pre> | List options |
```
Comment (scottinet) [inline doc/3/controllers/collection/list/index.md:28]: I think we should get rid of these optional parameters: https://github.com/kuzzleio/kuzzle/pull/1592

If you don't, then the default value for the `size` parameter should be `null` from this SDK's point of view.

Comment (Yoann-Abbes) [reply doc/3/controllers/collection/list/index.md]: I've didnt see that, thanks

**Pull request merged**